sonar-scanner-engine/src/main/java/org/sonar/scanner/genericcoverage/GenericCoverageSensor.java

```diff
@@ -0,0 +1,126 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.scanner.genericcoverage;
+
+import com.google.common.collect.ImmutableList;
+import java.io.File;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.api.batch.Initializer;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.config.PropertyDefinition;
+import org.sonar.api.config.Settings;
+import org.sonar.api.resources.Qualifiers;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import static org.sonar.api.CoreProperties.CATEGORY_CODE_COVERAGE;
+
+public class GenericCoverageSensor extends Initializer implements Sensor {
+
+  private static final Logger LOG = Loggers.get(GenericCoverageSensor.class);
+
+  private static final String REPORT_PATH_PROPERTY_KEY = "sonar.coverageReportPaths";
+  /**
+   * @deprecated since 6.2
+   */
+  @Deprecated
+  private static final String OLD_REPORT_PATH_PROPERTY_KEY = "sonar.genericcoverage.reportPath";
+  /**
+   * @deprecated since 6.2
+   */
+  @Deprecated
+  private static final String OLD_COVERAGE_REPORT_PATHS_PROPERTY_KEY = "sonar.genericcoverage.reportPaths";
+  /**
+   * @deprecated since 6.2
+   */
+  @Deprecated
+  private static final String OLD_IT_COVERAGE_REPORT_PATHS_PROPERTY_KEY = "sonar.genericcoverage.itReportPaths";
+  /**
+   * @deprecated since 6.2
+   */
+  @Deprecated
+  private static final String OLD_OVERALL_COVERAGE_REPORT_PATHS_PROPERTY_KEY = "sonar.genericcoverage.overallReportPaths";
+
+  private final Settings settings;
+
+  public GenericCoverageSensor(Settings settings) {
+    this.settings = settings;
+  }
+
+  public static ImmutableList<PropertyDefinition> properties() {
+    return ImmutableList.of(
+
+      PropertyDefinition.builder(REPORT_PATH_PROPERTY_KEY)
+        .name("Coverage report paths")
+        .description("List of comma-separated paths (absolute or relative) containing coverage report.")
+        .category(CATEGORY_CODE_COVERAGE)
+        .onQualifiers(Qualifiers.PROJECT)
+        .deprecatedKey(OLD_COVERAGE_REPORT_PATHS_PROPERTY_KEY)
+        .build());
+
+  }
+
+  @Override
+  public void execute() {
+    Set<String> reportPaths = new LinkedHashSet<>();
+    reportPaths.addAll(Arrays.asList(settings.getStringArray(REPORT_PATH_PROPERTY_KEY)));
+    loadDeprecated(reportPaths, OLD_REPORT_PATH_PROPERTY_KEY);
+    loadDeprecated(reportPaths, OLD_IT_COVERAGE_REPORT_PATHS_PROPERTY_KEY);
+    loadDeprecated(reportPaths, OLD_OVERALL_COVERAGE_REPORT_PATHS_PROPERTY_KEY);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "OLD_OVERALL_COVERAGE_REPORT_PATHS_PROPERTY_KEY"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)